### ISSUE TITLE
about learning rate decay in training

### ISSUE THREAD

**Issue by guolinke** [deepmind/alphafold#380]
from section 1.11.3 in Suppl. Material:
> We linearly increase (warm-up) the learning rate over the first 0.128 · 10^6 samples, and further decrease the learning rate by a factor of 0.95 after 6.4 · 10^6 samples.

as the batch size is 128, it is clear that the warmup steps is 1k.
However, for the decay,  `6.4 · 10^6 samples` is `50k` steps, does that mean the learning rate between `[1k, 50k]` steps is always `1e-3`, and suddenly decay to `0.95e-3` after 50k-step? or it is exponentially decaying from `1e-3` to `0.95e-3`?

**Comment by andycowie** [comment]
We exponentially decrease the learning rate from 50k steps (6.4*10^6 samples) with formula 1e-3*0.95**(steps//50000), halving by 700,000 steps.

This is a decay constant of -ln(0.95). Note that we discretise the exponent by using integer division, so that the learning rate decreases as a staircase function.

**Comment by guolinke** [comment]
Thank you so much @andycowie. 
I am confused about "700,000 steps". From table 4, the training only takes `80,000` steps (`10*10^6` samples). Is the training actually takes `800,000` steps, or the decay step is actually `5,000`?

**Comment by andycowie** [comment]
Good point - The 700,000 steps was only a hypothetical example, to show the halving time of the exponential decay. Since the training stops at around 80,000 steps, the effect is as you originally specify - the learning rate between `[1k, 50k]` steps is always `1e-3`, and suddenly decays to `0.95e-3` after 50k-step.
